src/bench/merkle_root.cpp

```diff
@@ -7,21 +7,34 @@
 #include <random.h>
 #include <uint256.h>
 
+#include <cassert>
 #include <vector>
 
 static void MerkleRoot(benchmark::Bench& bench)
 {
-    FastRandomContext rng(true);
-    std::vector<uint256> leaves;
-    leaves.resize(9001);
-    for (auto& item : leaves) {
+    FastRandomContext rng{/*fDeterministic=*/true};
+
+    std::vector<uint256> hashes{};
+    hashes.resize(9001);
```

Comment:
In the commit message of 8af0a3a72c8af781bebed7289657c637220df893 `validation: pre-reserve leaves to prevent reallocs with odd vtx count`:

> ... doubling its capacity (allocating 3x the necessary memory)

Why 3x?